### PR TITLE
Clarify perldoc bless per #18713

### PR DESCRIPTION
**Pull request by joemcmahon**
- Specifically calls out the possible invocations
 - Breaks out the specific use cases and why to use them
 - Notes the things to avoid and why in clearer language
 - Update .mailmap so that tests all pass

### CODE REVIEW

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod]: This is not correct - the referent is marked as an object, not the reference.

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod]: as above.

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod]: as above.

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod]: as above.

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod:990]: this is going to fail podcheck as the lines need to be under 78 characters when formatted (they will end up needing to be shorter since this is in the middle of a list in perlfunc).

File: pod/perlfunc.pod
Comment (joemcmahon) [reply pod/perlfunc.pod]: Changed to to `the referred-to item`.

File: pod/perlfunc.pod
Comment (joemcmahon) [reply pod/perlfunc.pod]: Changed to to `the referred-to item`.

File: pod/perlfunc.pod
Comment (joemcmahon) [reply pod/perlfunc.pod]: Changed to to `the referred-to item`.

File: pod/perlfunc.pod
Comment (joemcmahon) [reply pod/perlfunc.pod]: Got it -- I'm changing all these to `the referred-to item`. Fixed here.

File: pod/perlfunc.pod
Comment (tonycoz) [reply pod/perlfunc.pod:990]: You marked this resolved, but there's still very long lines.

Comment (joemcmahon) [reply pod/perlfunc.pod:990]: I ran it through podcheck without issues, but I'll shorten everything to 64 characters.

Comment (joemcmahon) [reply pod/perlfunc.pod:990]: I see the issue; there are lines other than this specific one that are longer that that; checking every line.

**Comment by khwilliamson** [comment]
On 10/23/22 21:47, Joe McMahon wrote:
> ***@***.**** commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In pod/perlfunc.pod 
> <https://github.com/Perl/perl5/pull/20428#discussion_r1002862026>:
> 
>> +
> +    bless $ref;
> +
> +If C<bless> is used without its second argument, the object is created in the current package. The second argument should always be supplied if a derived class might inherit a method executing C<bless>. Because it is a potential source of bugs, one-argument C<bless> is discouraged.
> +
> +=back
> +
> +See L<perlobj> for more about the blessing (and blessings) of objects.
> +
> +L<C<bless>|/bless REF,CLASSNAME> returns its first argument, the supplied reference, as the value of the function; since C<bless> is commonly the last thing executed in constructors, this returns the reference to the object by default and allows the caller to immediately use this returned object in method calls.
> +
> +C<CLASSNAME> should always be a mixed-case name, as all-uppercase and all-lowercase names are meant to be used only for Perl builtin types and pragmas, respectively. Avoid creating all-uppercase or all-lowercase package names to prevent confusion.
> +
> +Also avoid <Cbless>ing things into the class name C<0>; this will cause code which (erroneously) checks the result of L<C<ref>|/ref EXPR> to see if a reference is C<bless>ed to fail, as "0", a falsy value, is returned.
> +
> +See L<perlmod/"Perl Modules"> for more details.
> 
> I ran it through podcheck without issues, but I'll shorten everything to 
> 64 characters.
> 

podcheck only cares about lines that are to be output verbatim.  Those 
begin with a blank space in the first column.

The length of all others is irrelevant; the pod formatters, since 
they're not verbatim, will typically wrap them at the displayed width.

File: pod/perlfunc.pod
Comment (joemcmahon) [reply pod/perlfunc.pod:990]: All lines trimmed to 64 characters, plus or minus a few. The really long lines are all definitely gone.

**Comment by joemcmahon** [comment]
@khwilliamson Re wrapping, I was pretty sure that was the case, but there was no point arguing about it. In any case, themost recent version of the new text renders properly and is quite a bit clearer and more prescriptive of good practice.

**Comment by demerphq** [comment]
I resolved the conflict in .mailmap and forced pushed this for you @joemcmahon. I think the new wording is good and I will merge if it passes CI.

**Review by demerphq — APPROVED** [review]

**Comment by demerphq** [comment]
@Grinnz you have any follow up on this before i merge it?

**Comment by demerphq** [comment]
And forced push again with the correct fix. :-)

**Review by Grinnz — APPROVED** [review]

**Comment by demerphq** [comment]
Thanks, applied.

**Pull request merged**
